File: src/main/java/thkoeln/dungeon/eventconsumer/game/GameStatusEvent.java
package thkoeln.dungeon.eventconsumer.game;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.messaging.MessageHeaders;
import thkoeln.dungeon.eventconsumer.core.AbstractEvent;
import thkoeln.dungeon.game.domain.GameStatus;

import javax.persistence.Entity;
import java.util.UUID;

@Entity
@Getter
@Setter
@NoArgsConstructor ( access = AccessLevel.PROTECTED )
public class GameStatusEvent extends AbstractEvent {
    private GameStatus gameStatus;
    private UUID gameId;

    private static final String TYPE_KEY = "type";
    private static final String GAME_ID_KEY = "gameId";

    public GameStatusEvent( MessageHeaders messageHeaders, GameStatusEventPayload gameStatusEventPayload ) {
        super( messageHeaders );
        setGameStatus( gameStatusEventPayload.gameStatus() );
        setGameId( gameStatusEventPayload.gameId() );
    }
}

File: src/main/java/thkoeln/dungeon/eventconsumer/game/GameStatusEvent.kt
package thkoeln.dungeon.eventconsumer.game

import lombok.*
import org.springframework.messaging.MessageHeaders
import thkoeln.dungeon.eventconsumer.core.AbstractEvent
import thkoeln.dungeon.game.domain.GameStatus
import java.util.*
import javax.persistence.Entity

@Entity
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
class GameStatusEvent(messageHeaders: MessageHeaders, gameStatusEventPayload: GameStatusEventPayload) :
    AbstractEvent(messageHeaders) {
    var gameStatus: GameStatus? = null
    var gameId: UUID? = null

    init {
        this.gameStatus = gameStatusEventPayload.gameStatus
        this.gameId = gameStatusEventPayload.gameId
    }

    companion object {
        private const val TYPE_KEY = "type"
        private const val GAME_ID_KEY = "gameId"
    }
}